docs/README.md

```diff
@@ -0,0 +1,37 @@
+# The Shadow Simulator
```

Comment:
I think we need a `README.md` in the base of the repo, because many people will land on https://github.com/shadow/shadow and we want a file there to explain what the project is about. It's an extra click to get to `docs/`.

I think we should move this README back to the base, move the `contents` section to a new docs/README.md file, and then have the base README link to the docs README.

Reply:
> Is it possible to keep consistent with our rebase strategy to avoid merge commits and keep a clean commit history? Or is there some reason that won't work here?

Yeah, I'd previously tried rebasing everything on top or even cherry picking everything on top, but kept running into conflicts. I think part of the issue might be that the wiki itself has a branches and merges in the history. I'll poke a bit more though and see what I can do.

> I think we need a README.md in the base of the repo

If there's no README.md in the root, github will render docs/README.md instead. e.g. see https://github.com/sporksmith/shadow/tree/merge-rewritten-wiki. You're proposed structure sounds fine to me too, though. I'll go ahead and give that a shot.

Reply:
Took another stab at this.

Indeed the tricky part is that rebase seems to want to serialize the wiki history, which results in having to (re)resolve some merge conflicts. I went through and made a best effort of doing so. I'm not 100% sure I got it right, but at least the history is mostly preserved. I also validated that the end-state matched the wiki repo before re-applying my fixups.

I also left the two READMEs separate as-requested.